src/commands/executeFunction/executeEventGridFunction.ts

```diff
@@ -0,0 +1,156 @@
+        (
+            await context.ui.showQuickPick(eventTypePicks, {
+                placeHolder: localize('selectEventType', 'Select the event type'),
+                stepName: 'eventType',
+            })
+        ).data ?? 'blob_created.json';
+
+    // Get selected contents of sample request
+    const selectedFileUrl = sampleFiles.find((fileMetadata) => fileMetadata.name === selectedFileName)?.download_url || sampleFiles[0].download_url;
+    const selectedFileContents: {} = await feedUtils.getJsonFeed(context, selectedFileUrl);
+
+    // Prompt for whether to send or modify the sample request
+    const shouldModify: boolean =
+        (
+            await context.ui.showQuickPick(
+                [
+                    {
+                        label: localize('sendSample', 'Send sample request'),
+                        data: false,
+                    },
+                    {
+                        label: localize('modifySample', 'Modify sample request'),
+                        data: true,
+                    },
+                ],
+                {
+                    placeHolder: 'Would you like to send the sample request or modify it first?',
+                    stepName: 'modifyOrSendSample',
+                },
+            )
+        ).data || false;
+
+    // Execute function with sample data directly if user chooses not to modify
+    if (!shouldModify) {
+        return executeFunctionWithInput(context, selectedFileContents, node);
+    }
+
+    // Create a temp file with the sample request & open in new window
+    const tempFilePath: string = await createTempSampleFile(eventSource, selectedFileName, selectedFileContents);
+    const document: vscode.TextDocument = await vscode.workspace.openTextDocument(tempFilePath);
+    await vscode.window.showTextDocument(document, {
+        preview: false,
+    });
+
+    // Request will be sent when the user clicks on the button or on the codelens link
+
+    // Set a listener to delete the temp file after it's closed
+    await new Promise<void>((resolve, reject) => {
+        const disposable = vscode.workspace.onDidCloseTextDocument(async (closedDocument) => {
+            if (closedDocument.fileName === document.fileName) {
+                try {
+                    await fs.unlink(tempFilePath);
+                    resolve();
+                } catch (error) {
+                    reject(error);
+                } finally {
+                    disposable.dispose();
+                }
+            }
+        });
+    });
+}
+
+async function createTempSampleFile(eventSource: string, fileName: string, contents: {}): Promise<string> {
+    const samplesDirPath = await createSamplesDirIfNotExists(eventSource);
+    const sampleFileName = fileName.replace(/\.json$/, '.eventgrid.json');
+    const filePath: string = path.join(samplesDirPath, sampleFileName);
+
+    await fs.writeFile(filePath, JSON.stringify(contents, undefined, 2));
```

Comment:
Use `AzExtFsExtra.writeFile` for any file system provider calls. `AzExtFsExtra` uses all of the `vscode.workspace.fs` APIs so this will help our extensions stay compatible with all the different VS Code platforms.

Reply:
Done